cache/types.ts

```diff
@@ -0,0 +1,15 @@
+import type { Operation, Stream } from "npm:effection@4.0.0-alpha.3";
+
+export interface Cache {
+  location: URL;
+  write(key: string, data: unknown): Operation<void>;
+  append(key: string, data: unknown): Operation<void>;
+  read<T>(key: string): Operation<Stream<T, unknown>>;
```

Comment:
What is the value in having a stream (itself an operation) wrapped in an operation? In other words: an operation that yields an operation that yields a subscription. Why not just make it a stream?

Reply:
This type is a side effection of `Deno.open(file)` returning a promise, so I need an operation to use `call`, but I guess I could probably return a subscription directly. Need to look into that.

Reply:
So when you call read, are you basically iterating over the results?

Reply:
Yes